### DISCUSSION TITLE
Cannot establish connection using ipv6 address

### DISCUSSION THREAD

**Discussion by 3157696** [strongswan/strongswan#219] [Q&A] [answered]
Hello
There are two hosts A and B, and their configurations are the same 
ipsec.conf
```
conn a
	#left=2001:8a55:2417:a0e0:a00:27ff:fe22:9c91
	left=%any6
	#left=192.168.1.123
	right=2001:8a55:2417:a0e0:a00:27ff:fe44:73f7
	#right=192.168.1.108

	leftid=a
	rightid=b
	leftauth=psk
	rightauth=psk
	auto=start
	leftfirewall=yes
```
A can reach B through ping/traceroute using ipv6 
Can establish connection using ipv4 address 
ip6tables is ACCEPT

**Q1**
When set to left=2001:....  instead of %any6 
**Why  say left nor right host is our side** 
```
10[CFG] received stroke: add connection 'a'
10[CFG] left nor right host is our side, assuming left=local
10[CFG] added configuration 'a'
12[CFG] received stroke: initiate 'a'
12[IKE] initiating IKE_SA a[1] to 2409:8a55:2417:a0e0:a00:27ff:fe44:73f7
12[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
12[NET] sending packet: from 2001:8a55:2417:a0e0:a00:27ff:fe22:9c91[500] to 2001:8a55:2417:a0e0:a00:27ff:fe44:73f7[500] (296 bytes)
04[NET] error writing to socket: Invalid argument
14[IKE] retransmit 1 of request with message ID 0
14[NET] sending packet: from 2001:8a55:2417:a0e0:a00:27ff:fe22:9c91[500] to 2001:8a55:2417:a0e0:a00:27ff:fe44:73f7[500] (296 bytes)
04[NET] error writing to socket: Invalid argument
```
**Q2**
The connection log is as follows   when set to left= %any6
```
10[CFG] received stroke: add connection 'a'
10[CFG] added configuration 'a'
12[CFG] received stroke: initiate 'a'
12[IKE] initiating IKE_SA a[1] to 2001:8a55:2417:a0e0:a00:27ff:fe44:73f7
......
12[IKE] peer not responding, trying again (3/3)
12[IKE] initiating IKE_SA a[1] to 2001:8a55:2417:a0e0:a00:27ff:fe44:73f7
12[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
12[NET] sending packet: from 2001:8a55:2416:4740:a00:27ff:fe22:9c91[500] to 2001:8a55:2417:a0e0:a00:27ff:fe44:73f7[500] (296 bytes)
14[IKE] retransmit 1 of request with message ID 0
14[NET] sending packet: from 2001:8a55:2416:4740:a00:27ff:fe22:9c91[500] to 2001:8a55:2417:a0e0:a00:27ff:fe44:73f7[500] (296 bytes)
...
08[IKE] retransmit 5 of request with message ID 0
08[NET] sending packet: from 2001:8a55:2416:4740:a00:27ff:fe22:9c91[500] to 2001:8a55:2417:a0e0:a00:27ff:fe44:73f7[500] (296 bytes)
09[IKE] giving up after 5 retransmits
09[IKE] establishing IKE_SA failed, peer not responding
```
Execute _tcpdump ip6 -i any and port 500_ 
At this time, host B cannot receive 500 packets from A 
Is it because host A doesn't know who is B  or is rejected by B ?

Thanks for your help.

**Comment by tobiasbrunner** [comment, marked as answer]
>  **Why say left nor right host is our side**

Probably because that's not an IP address on that host. As you can see in the log, you can't send messages from that IP address either.



>  The connection log is as follows when set to left= %any6

You can clearly see that the message is sent from a different source address (`2001:8a55:2416:4740:a00:27ff:fe22:9c91` and not `2001:8a55:2417:a0e0:a00:27ff:fe22:9c91`, which you configured). Perhaps you have the same issue on the other end (or with the remote address). Make sure you configure the correct IP addresses (or let the daemon figure out the source address itself).

>  Is it because host A doesn't know who is B or is rejected by B ?

Maybe a configuration problem (if you set `right` to an address that is not correct), or a firewall issue.

**Reply by 3157696** [reply]
Thanks for your help.
The problem is solved
I was too careless
The dhcp address changed and I didn’t find it
